tests/ulm/erc20/rust/src/encoder.rs

```diff
@@ -0,0 +1,102 @@
+// Encodes values according to the solidity ABI:
+// https://docs.soliditylang.org/en/latest/abi-spec.html
+//
+// To encode, first implement the Encodable trait for the types you want to
+// encode. Then do something like this:
+//
+// let mut encoder = Encoder::new();
+// encoder.add(value_1);
+// encoder.add(value_2);
+// ...
+// encoder.add(value_n);
+// let encoded = encoder.encode();
+
+use bytes::{Bytes, BytesMut, BufMut, Buf};
+
+use crate::unsigned::U256;
+
+pub enum EncodingType {
+    FixedSize = 1,
+    VariableSize = 2,
+}
+
+pub trait Encodable {
+    fn encode(&self) -> (EncodingType, Bytes);
+}
+
+impl Encodable for String {
+    fn encode(&self) -> (EncodingType, Bytes) {
+        let mut result = BytesMut::new();
```

Comment:
It seems like we could do a `BytesMut::with_capacity()` here as well, since we know the final size that this buffer will be.

Reply:
Done.